packages/starknet_provider/lib/src/model/invoke_transaction.dart

```diff
@@ -18,10 +18,18 @@ class InvokeTransactionRequest with _$InvokeTransactionRequest {
 }
 
 abstract class InvokeTransaction {
-  factory InvokeTransaction.fromJson(Map<String, Object?> json) =>
-      json['version'] == '0x01'
-          ? InvokeTransactionV1.fromJson(json)
-          : InvokeTransactionV0.fromJson(json);
+  factory InvokeTransaction.fromJson(Map<String, Object?> json) {
+    switch (json['version']) {
+      case '0x00':
+        return InvokeTransactionV0.fromJson(json);
+      case '0x01':
+        return InvokeTransactionV1.fromJson(json);
+      case '0x03':
+        return InvokeTransactionV3.fromJson(json);
+      default:
+        throw ArgumentError('Unsupported transaction version:');
+    }
+  }
```

Comment:
According to JSON RPC specifications it should be `0x0`, `0x1`, '0x3`.
I think we have previously used `0x01` due to an implementation not following the JSON RPC specifcation.
So I suggest to handle both cases: `0x0v` and `0xv`